### DISCUSSION TITLE
Windows

### DISCUSSION THREAD

**Discussion by adesutherland** [adesutherland/CREXX#293] [General]
Have been struggling to get Windows building consistently with my Windows VM and the GITHUB runner - usual Windows "fun and games"!

This included struggling with the bundled version of Mingw in Clion 

Anyway in the end I cleared down my machine and used the wonderful MSYS2 (https://www.msys2.org/) - this includes pacman, a package manager.

Anyway it easily provides a sane and up to date GCC & CLANG toolchain/environment. You easily get CLION to use it, and I can use it in the Github Windows runner - so we have the same versions there too.

So @Peter-Jacob @rvjansen this might be the standard we use to have consistent Windows build environment?

Also thinking about future releases - we should make CREXX a MSYS2 package - and perhaps in the short term as a POC/test package Regina to MSYS2 - this would give a much simpler install for windows users.

What do you think?

**Comment by rvjansen** [comment]
I have no real machines running windows, but I had to replace Virtual Box by Parallels when moving to the M1 (I only have 1 power CPU left and one Intel but they are rarely switched on.) In Parallels I can run Windows 11 (M1 native), and when I install MSYS2, it immediately freezes. But don't let that discourage you, it probably will work eventually. Another image with Clion does work however. The MS compiler stumbles over warning switches when building CREXX. The ideal installer - in my opinion - is the one that runs a script after a curl (like rustup) and just decompresses the needed code somewhere. Windows installers were a lot of headache over the years; they should not assume a specific environment.

**Comment by Peter-Jacob** [comment]
I happily run with MINGW64. for BREXX we use CYGWIN, I think because of TCP support.
![Screenshot 2022-05-26 19 15 05](https://user-images.githubusercontent.com/46247351/170540656-9996f7fc-b51a-413a-85dd-af32a6fb6bbe.png)
